http-server-netty/build.gradle

```diff
@@ -26,6 +26,8 @@ dependencies {
     api project(":http-netty")
     api libs.managed.netty.codec.http
 
+    implementation project(":crac")
```

Comment:
We cannot introduce a new dependency to support CraC. This would need to be `compileOnly` and the feature implemented in such a way as it doesn't need to be present on the classpath

Reply:
I was trying to do that, but CRaC requires that you register a class with it's core registry...  And that class must implement `Resource`

I couldn't think of a way around this restriction...  And of course, as soon as we implement Resource, we need the new dependency on the classpath 🤔

Do we have an example somewhere in the codebase where this is achieved?  Or do you have any ideas on a direction to go in?

Reply:
The org.crac library is a 9k shim library that;

> ...is designed to provide smooth CRaC adoption. Users of the library can build against and use CRaC API on Java runtimes with jdk.crac, javax.crac, or without any implementation.

(_[source](https://github.com/CRaC/org.crac/)_)

Reply:
Hmm yeah even if it is a shim, personally I am opposed to introducing any other third party dependencies. I suggest we bring this up with the CraC folks so they can suggest solutions to our issue

Reply:
👍  I'll try a few things the remainder of today, and reach out to them if I can't find a way through

Reply:
Found a way 🎉

I think 🤔

Pushed it, and testing it out locally... I'll fire up.an EC2 box with their JDK tomorrow, and manually check it works